web-api/openapi/schemas/document.yml

```diff
@@ -11,29 +11,32 @@ DocumentPropertyId:
 DocumentPropertyString:
   description: |
     Arbitrary string property that can be attached to a document.
-    Enclosing whitespace will be trimmed.
     The length constraints are in bytes, not characters.
   type: string
   pattern: '^[^\x00]+$'
-  minLength: 1
+  minLength: 0
   maxLength: 2048
 
+DocumentPropertyArrayString:
+  description: | 
+    Array of non-null string.
+  type: array
+  minItems: 0
+  items:
+    $ref: '#/DocumentPropertyString'
+
 DocumentProperty:
   description: | 
-    Arbitrary property that can be attached to a document.
-    When a property is set to `null`, it is treated as if that property had no values.
+    Arbitrary data that can be attached to a document. A subset of JSON is supported.
+    When a property is set to null, or an empty array it is treated as though that property has no values.
+    This means that they will not appear in searches with filter on this properties.
```

Comment:
:art: phrasing
```suggestion
    When a property is set to null or an empty array, it is treated as if that property had no values.
    This means that they won't appear in searches with filters on these properties.
```

Reply:
I should trust less the phrasing from ES documentation :grimacing: